src/extension.ts

```diff
@@ -8,13 +9,24 @@ import { Util } from './util';
 import { RecordingOptions } from './types';
 import { Config } from './config';
 
-export function activate(context: vscode.ExtensionContext) {
+export async function activate(context: vscode.ExtensionContext) {
 
   Util.context = context;
 
   const recorder = new Recorder();
   const status = new RecordingStatus();
 
+  const vslsApi = await vsls.getApi();
```

Comment:
Swap the order of the checks, making sure `Config.getAutoRecordLiveShare()` is true before checking anything else.

Also, do a dynamic `import` inside the if block for the `vsls` library so we don't have to load it, if the flag is not set to true.

Reply:
Wrap the logic in a function and move to the end of the block.  I generally try to order functions by dependencies, meaning all of the dependent functions come first in the block.

Reply:
Done!